sklearn/manifold/_isomap.py

```diff
@@ -159,6 +161,51 @@ class Isomap(_ClassNamePrefixFeaturesOutMixin, TransformerMixin, BaseEstimator):
     (100, 2)
     """
 
+    _parameter_constraints = {
+        "n_neighbors": [Interval(Integral, 0, None, closed="left"), None],
+        "radius": [Interval(Real, 0.0, None, closed="left"), None],
+        "n_components": [Interval(Integral, 1, None, closed="left"), None],
+        "eigen_solver": [StrOptions({"auto", "dense", "arpack"})],
+        "tol": [Interval(Real, 0.0, None, closed="left")],
+        "max_iter": [Interval(Integral, 0, None, closed="left"), None],
+        "path_method": [StrOptions({"auto", "FW", "D"})],
+        "neighbors_algorithm": [StrOptions({"auto", "brute", "kd_tree", "ball_tree"})],
+        "n_jobs": [Integral, None],
+        "metric": [
+            StrOptions(
+                {
+                    "minkowski",
+                    "cityblock",
+                    "cosine",
+                    "euclidean",
+                    "l1",
+                    "l2",
+                    "manhattan",
+                    "braycurtis",
+                    "canberra",
+                    "chebyshev",
+                    "correlation",
+                    "dice",
+                    "hamming",
+                    "jaccard",
+                    "mahalanobis",
+                    "minkowski",
+                    "rogerstanimoto",
+                    "russellrao",
+                    "seuclidean",
+                    "sokalmichener",
+                    "sokalsneath",
+                    "sqeuclidean",
+                    "yule",
+                    "precomputed",
+                }
+            ),
+            callable,
+        ],
+        "p": [Interval(Integral, 0, 1, closed="neither")],
```

Comment:
In the results for the testing checks in GitHub, I'm seeing that this parameter is not correctly defined. By default, the value is 2, which is outside of the range you defined. Sometimes, I've noticed that the tests fail on a parameter that seems correct, but it is actually another one that's causing the issue.

Reply:
Thank you for catching this!